test/val/val_image_test.cpp

```diff
@@ -8046,6 +8038,302 @@ TEST_F(ValidateImage, ImageMSArray_ArrayedTypeDoesNotRequireCapability) {
+    OpDecorate %uniform_image DescriptorSet 0
+    OpDecorate %uniform_image Binding 0
+     %void = OpTypeVoid
+     %func = OpTypeFunction %void
+      %u32 = OpTypeInt 32 0
+    %u32_0 = OpConstant %u32 0
+    %u32_1 = OpConstant %u32 1
+  %u32vec2 = OpTypeVector %u32 2
+  %u32vec4 = OpTypeVector %u32 4
+%u32vec2_01 = OpConstantComposite %u32vec2 %u32_0 %u32_1
+ %img_type = OpTypeImage %u32 2D 0 0 0 1 R32i
+%ptr_image = OpTypePointer UniformConstant %img_type
+%uniform_image = OpVariable %ptr_image UniformConstant
+     %main = OpFunction %void None %func
+    %label = OpLabel
+      %img = OpLoad %img_type %uniform_image
+     %res1 = OpImageFetch %u32vec4 %img %u32vec2_01 ZeroExtend
+    OpReturn
+    OpFunctionEnd
+)";
+
+  CompileSuccessfully(code.c_str(), SPV_ENV_VULKAN_1_2);
+  ASSERT_EQ(SPV_ERROR_INVALID_DATA, ValidateInstructions(SPV_ENV_VULKAN_1_2));
+  EXPECT_THAT(getDiagnosticString(),
+              AnyVUID("VUID-StandaloneSpirv-Image-04965"));
+  EXPECT_THAT(
+      getDiagnosticString(),
+      HasSubstr("Image Format signedness does not match Sample Type operand "
+                "including possible SignExtend or ZeroExtend operand"));
+}
+
+TEST_F(ValidateImage, TypeImageVulkanStorageZeroExtendFloat) {
+  // use ZeroExtend on Float image
+  const std::string code = R"(
+            OpCapability Shader
+            OpCapability StorageImageWriteWithoutFormat
+            OpMemoryModel Logical GLSL450
+            OpEntryPoint GLCompute %main "main" %var
+            OpExecutionMode %main LocalSize 1 1 1
+            OpDecorate %var DescriptorSet 0
+            OpDecorate %var Binding 0
+            OpDecorate %var NonReadable
+    %void = OpTypeVoid
+    %func = OpTypeFunction %void
+    %int = OpTypeInt 32 1
+    %float = OpTypeFloat 32
+    %image = OpTypeImage %float 2D 0 0 0 2 Unknown
+    %ptr = OpTypePointer UniformConstant %image
+    %var = OpVariable %ptr UniformConstant
+    %v2int = OpTypeVector %int 2
+    %int_1 = OpConstant %int 1
+    %coord = OpConstantComposite %v2int %int_1 %int_1
+    %v3float = OpTypeVector %float 3
+    %float_1 = OpConstant %float 1
+%texelU3 = OpConstantComposite %v3float %float_1 %float_1 %float_1
+    %main = OpFunction %void None %func
+    %label = OpLabel
+    %load = OpLoad %image %var
+            OpImageWrite %load %coord %texelU3 ZeroExtend
+            OpReturn
+            OpFunctionEnd
+)";
+
+  CompileSuccessfully(code.c_str(), SPV_ENV_VULKAN_1_2);
+  ASSERT_EQ(SPV_ERROR_INVALID_DATA, ValidateInstructions(SPV_ENV_VULKAN_1_2));
+  EXPECT_THAT(getDiagnosticString(),
+              AnyVUID("VUID-StandaloneSpirv-Image-04965"));
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("Can't use SignExtend or ZeroExtend on float images"));
+}
+
+TEST_F(ValidateImage, BothSignExtendZeroExtendTogether) {
+  const std::string body = R"(
+%img = OpLoad %type_image_s32_2d_0002 %uniform_image_s32_2d_0002
+%res1 = OpImageRead %s32vec4 %img %u32vec2_01 SignExtend ZeroExtend
```

Comment:
I think this is written incorrectly. It should be:
```suggestion
%res1 = OpImageRead %s32vec4 %img %u32vec2_01 SignExtend|ZeroExtend
```
Right now it is two operands and the parser is complaining. Though I admit it seems strange that both would be allowed, I can't find where it is forbidden. Maybe that should be a spec issue.

Reply:
going to remove this test and come back later with it once we get spec text (also it is slightly out of scope of this PR)